waltz/imploder
# Switch from Travis CI to GitHub Actions (#27)

* Try installing a few deps

* Install Node, Ruby, and attach a Postgres instance

* Match the installed version of ruby with the one in ruby-version

* Use a clever method for finding the current ruby version

https://github.com/actions/setup-ruby/issues/31#issuecomment-561144368

* Oops, gotta checkout the code before reading the version

* Bump to the latest patch version of Ruby

* Bump the Gemfile too

* Health check the Postgres instance and split out the bundle step

* use the latest Postgres, reformat the heallth check

* Point Rails at the service container

* Switch back to port mapping postgres to the local machine

* Manually set up the postgres instance and use DATABASE_URL for telling Rails where to look

* Deploy using GitHub workflows

* Use secrets store

* Setup both api. and git. for Heroku

* Specify the Heroku app to deploy to

* Double check current config

* debug heroku auth

* Work that netrc

* try and preserve newlines when writing netrc

* Be extra specific with the GH branch we want to push

* Only deploy on master

* I read the docs

* Install specific node versions

* Cache Node Modules runs

* Cache Ruby dependencies

* Ditch Travis

Switch the build badge from Travis to GH Actions

ayy

wow

Clobber the remote ref

Fix a Gemfile Ruby version issue

Fetch the whole repository because Heroku wants it

Keep the Heroku app names secret and Deploy to Production
